### PR TITLE
🎓 Create model for mentor education

### LINKED ISSUE #7: Model for education information for profile
#### Description
As listed on the doc, create a model for a mentor's education information. It should look like:

> Education 
> - Highest level of education (dropdown) - StringField
> - Major(s) - List of String
> - School - String
> - Graduation date - Int

Include constructor (`__init__`) and `__repr__` functions

#### Files
`backend/api/models/[namehere].py`

#### Tasks
- [ ] Create the model!

### PR DESCRIPTION
**Pull request by Feeshay**
## Status:
:rocket: Ready

## Description :star2:
* Creates a model for a mentor's education as described on our DB structure doc
* MongoDB API reference: https://docs.mongoengine.org/apireference.html
* Fixes #7 
 
## TODOs :star:
- [ ] Integrate into full mentor model

### CODE REVIEW

File: backend/api/models/Education.py [hunk 1]
```diff
@@ -0,0 +1,23 @@
+from api.core import Mixin
+from .base import db
+from mongoengine import *
+
+
+class Email(EmbeddedDocument, Mixin):
```
Comment (chyku) [inline backend/api/models/Education.py:6]: rename this lol

File: backend/api/models/Education.py [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+    def __init__(self, educationLevel, majors, school, graduationYear):
+        self.educationLevel = education_level
+        self.majors = majors
+        self.school = school
+        self.graduationYear = graduationYear
+
+    def __repr__(self):
+        return f"""<Highest degree {self.educationLevel}, 
+                    majors {self.majors}, school {self.school}, 
+                    graduation date {self.graduationYear}>"""
```
Comment (chyku) [inline backend/api/models/Education.py:23]: 3 quotes? Also include the class name here and have a colon after the field name like "majors: {self.majors}"

**Review by chyku — CHANGES_REQUESTED** [review]
small things

File: backend/api/models/Education.py (hunk 1 above)
Comment (Feeshay) [reply backend/api/models/Education.py:6]: f

File: backend/api/models/Education.py (hunk 2 above)
Comment (Feeshay) [reply backend/api/models/Education.py]: 3 quotes is for multiline strings but I could also just put everything into 1 line

Comment (chyku) [reply backend/api/models/Education.py]: oh lol oop

**Review by Leundai — CHANGES_REQUESTED** [review]
within init.py of the folder, models make sure to also include MentorProfile in there. like so. Not too sure if that is needed though
image

File: backend/api/models/Education.py [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+from mongoengine import *
+
+
+class Education(EmbeddedDocument, Mixin):
+    """Education embedded within Mentor."""
+
+    education_level = StringField()
+    majors = ListField(StringField())
+    school = StringField()
+    graduation_year = IntField()
```
Comment (chyku) [inline backend/api/models/Education.py:12]: Should also mark which fields are required (i.e. `title = StringField(max_length=200, required=True)`)

**Review by chyku — CHANGES_REQUESTED** [review]

File: backend/api/models/Education.py [hunk 4]
```diff
@@ -0,0 +1,23 @@
+from api.core import Mixin
+from .base import db
+from mongoengine import *
+
+
+class Education(EmbeddedDocument, Mixin):
+    """Education embedded within Mentor."""
+
+    education_level = StringField(max_length=200, required=True)
+    majors = ListField(StringField(), min_length=1, required=True)
```
Comment (chyku) [inline backend/api/models/Education.py:10]: it's not necessary to put max and min lengths for every field, just that required is true for these

**Review by chyku — CHANGES_REQUESTED** [review]

**Review by chyku — APPROVED** [review]
good stuff :book:

**Review by Lam7150 — APPROVED** [review]
beautiful, just like you Michael

**Review by Leundai — APPROVED** [review]
Awesome!

**Pull request merged**